src/plugins/messageStorage/text.js

```diff
@@ -46,24 +46,67 @@ class TextFileMessageStorage {
 		const time = moment(msg.time).utcOffset(tz).format(format);
 		let line = `[${time}] `;
 
-		if (msg.type === "message") {
-			// Format:
-			// [2014-01-01 00:00:00] <Arnold> Put that cookie down.. Now!!
-			line += `<${msg.from.nick}> ${msg.text}`;
-		} else {
-			// Format:
-			// [2014-01-01 00:00:00] * Arnold quit
-			line += `* ${msg.from.nick} `;
+		switch (msg.type) {
+		case "message":
+			// [2014-01-01 00:00:00] <@Arnold> Put that cookie down.. Now!!
+			line += `<${msg.from.mode}${msg.from.nick}> ${msg.text}`;
```

Comment:
I believe ZNC doesn't log with modes. Does PISG for example support modes in energymech log parser?

Reply:
You are right. I looked at PISG's individual log parsers, and I found one (IRCap) where the user mode is being removed from the nick. They don't do that for energymech, so I guess that's a no.

So you want to be compatible with something like PISG I take it?

Reply:
Yes please, PISG support would be highly appreciated!